src/ray/gcs/gcs_client/service_based_gcs_client.cc

```diff
@@ -147,6 +147,14 @@ void ServiceBasedGcsClient::PeriodicallyCheckGcsServerAddress() {
     }
   }
 
+  // If gcs service restart detector is not enabled, we will stop periodic check after
+  // the connection with gcs service is established.
+  if (!current_gcs_server_address_.first.empty() &&
```

Comment:
Is this the same behavior as before the original PR? I'm wondering because it looks like there could still be a race condition here:
1. GCS service starts.
2. GCS service stops.
3. Worker node starts, enters infinite loop waiting for GCS address even though GCS restart is disabled.

It would be best if we could make sure that the worker node eventually crashes if it can't find the GCS.